Mathlib/Init/Algebra/Classes.lean

```diff
@@ -142,56 +147,12 @@ class is_idempotent (α : Sort u) (f : α → α) : Prop :=
 (idempotent : ∀ a, f (f a) = f a)
 -/
 
-/-- `IsIrrefl X r` means the binary relation `r` on `X` is irreflexive (that is, `r x x` never
-holds). -/
-class IsIrrefl (α : Sort u) (r : α → α → Prop) : Prop where
-  irrefl : ∀ a, ¬r a a
-
-/-- `IsRefl X r` means the binary relation `r` on `X` is reflexive. -/
-class IsRefl (α : Sort u) (r : α → α → Prop) : Prop where
-  refl : ∀ a, r a a
-
-/-- `IsSymm X r` means the binary relation `r` on `X` is symmetric. -/
-class IsSymm (α : Sort u) (r : α → α → Prop) : Prop where
-  symm : ∀ a b, r a b → r b a
-
 /-- The opposite of a symmetric relation is symmetric. -/
 instance (priority := 100) isSymmOp_of_isSymm (α : Sort u) (r : α → α → Prop) [IsSymm α r] :
     IsSymmOp α Prop r where symm_op a b := propext <| Iff.intro (IsSymm.symm a b) (IsSymm.symm b a)
 
-/-- `IsAsymm X r` means that the binary relation `r` on `X` is asymmetric, that is,
-`r a b → ¬ r b a`. -/
-class IsAsymm (α : Sort u) (r : α → α → Prop) : Prop where
-  asymm : ∀ a b, r a b → ¬r b a
-
-/-- `IsAntisymm X r` means the binary relation `r` on `X` is antisymmetric. -/
-class IsAntisymm (α : Sort u) (r : α → α → Prop) : Prop where
-  antisymm : ∀ a b, r a b → r b a → a = b
-
-instance (priority := 100) IsAsymm.toIsAntisymm {α : Sort u} (r : α → α → Prop) [IsAsymm α r] :
-    IsAntisymm α r where
-  antisymm _ _ hx hy := (IsAsymm.asymm _ _ hx hy).elim
-
-/-- `IsTrans X r` means the binary relation `r` on `X` is transitive. -/
-class IsTrans (α : Sort u) (r : α → α → Prop) : Prop where
-  trans : ∀ a b c, r a b → r b c → r a c
-
-instance {α : Sort u} {r : α → α → Prop} [IsTrans α r] : Trans r r r :=
-  ⟨IsTrans.trans _ _ _⟩
-
-instance (priority := 100) {α : Sort u} {r : α → α → Prop} [Trans r r r] : IsTrans α r :=
-  ⟨fun _ _ _ => Trans.trans⟩
-
-/-- `IsTotal X r` means that the binary relation `r` on `X` is total, that is, that for any
-`x y : X` we have `r x y` or `r y x`. -/
-class IsTotal (α : Sort u) (r : α → α → Prop) : Prop where
-  total : ∀ a b, r a b ∨ r b a
-
-/-- `IsPreorder X r` means that the binary relation `r` on `X` is a pre-order, that is, reflexive
-and transitive. -/
-class IsPreorder (α : Sort u) (r : α → α → Prop) extends IsRefl α r, IsTrans α r : Prop
-
 /-- `IsTotalPreorder X r` means that the binary relation `r` on `X` is total and a preorder. -/
+@[deprecated (since := "2024-07-30")]
 class IsTotalPreorder (α : Sort u) (r : α → α → Prop) extends IsTrans α r, IsTotal α r : Prop
```

Comment:
Lots of things have been deprecated in this PR, with no explanation in the description as to the reasons. Even if they are the right choices, it's hard to understand why in all cases.

Reply:
I am deprecating all the classes that mathlib is not using. This is a long standing TODO in those Init files: Deciding what's useful and what's not. If you scroll up a bit, you will see that Mario has written a fairly detailed list of all those classes, with hints to which ones are useful to keep and which ones are not